src/PaddedViews.jl

```diff
@@ -160,7 +160,7 @@ Base.size(A::PaddedView) = map(length, axes(A))
 
 Base.parent(A::PaddedView) = A.data
 
-@inline function Base.getindex(A::PaddedView{T,N}, i::Vararg{Int,N}) where {T,N}
+Base.@propagate_inbounds function Base.getindex(A::PaddedView{T,N}, i::Vararg{Int,N})::T where {T,N}
```

Comment:
Unless there's an inference failure, adding a return-type declaration should have no impact. The `@propagate_inbounds` is more likely to account for the change.

Reply:
I had the same feeling until I see the benchmark result.

```julia
julia> A = reshape(collect(1:6), 2, 3);

julia> Ap = PaddedView(missing, A, (0:4, 0:4));

julia> @btime getindex($Ap, 1, 1);
  3.550 ns (0 allocations: 0 bytes) # without
  2.532 ns (0 allocations: 0 bytes) # with

julia> @btime getindex($Ap, 0, 0);
  2.254 ns (0 allocations: 0 bytes) # without
  1.977 ns (0 allocations: 0 bytes) # with

julia> @btime @inbounds getindex($Ap, 1, 1);
  2.197 ns (0 allocations: 0 bytes) # without
  1.975 ns (0 allocations: 0 bytes) # with

julia> @btime @inbounds getindex($Ap, 0, 0);
  1.706 ns (0 allocations: 0 bytes) # without
  1.697 ns (0 allocations: 0 bytes) # with
```

With `T` annotation it's always faster in the first case.

```julia
julia> @benchmark getindex($Ap, 1, 1)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     3.007 ns (0.00% GC)
  median time:      3.108 ns (0.00% GC)
  mean time:        3.180 ns (0.00% GC)
  maximum time:     31.781 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000

julia> @benchmark getindex($Ap, 1, 1)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     2.462 ns (0.00% GC)
  median time:      2.536 ns (0.00% GC)
  mean time:        2.532 ns (0.00% GC)
  maximum time:     21.364 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000
```

Reply:
For me, `@btime` is all over the map here:
```julia
julia> @btime getindex($Ap, 1, 1)
  2.949 ns (0 allocations: 0 bytes)
1

julia> @btime getindex($Ap, 1, 1)
  2.215 ns (0 allocations: 0 bytes)
1

julia> @btime getindex($Ap, 1, 1)
  2.705 ns (0 allocations: 0 bytes)
1

julia> @btime getindex($Ap, 1, 1)
  2.215 ns (0 allocations: 0 bytes)
1
```
All this without making any changes to the code.

This result basically *can't* be true. Check
```julia
get00(A) = A[0,0]
@code_typed get00(Ap)
```
for master and this PR. It has to be a benchmarking artifact.